x-pack/plugins/apm/common/environment_filter_values.ts

```diff
@@ -33,3 +33,13 @@ export const ENVIRONMENT_NOT_DEFINED = {
 export function getEnvironmentLabel(environment: string) {
   return environmentLabels[environment] || environment;
 }
+
+export function isEnvironmentEqualToQueryEnvironment(
+  environment?: string,
+  queryEnvironment?: string
+) {
+  const environmentToCheck = environment ?? ENVIRONMENT_NOT_DEFINED.value;
+  const queryEnvironmentToCheck = queryEnvironment ?? ENVIRONMENT_ALL.value;
+
+  return environmentToCheck === queryEnvironmentToCheck;
+}
```

Comment:
It looks like everywhere this function is called, the same ternary is repeated. Might make sense to move that logic in here:

```ts
export function getEnvironmentSomethingSomething(
  environment?: string,
  queryEnvironment?: string
) {
  const environmentToCheck = environment ?? ENVIRONMENT_NOT_DEFINED.value;
  const queryEnvironmentToCheck = queryEnvironment ?? ENVIRONMENT_ALL.value;

  return environmentToCheck === queryEnvironmentToCheck ? queryEnvironment : ENVIRONMENT_ALL.value;
}
```

Reply:
agreed 👍 great name as well

Reply:
Thanks, I thought you'd like it :D